tests/phpunit/tests/interactivity-api/wpInteractivityAPI.php

```diff
@@ -785,6 +789,9 @@ public function test_evaluate_value() {
 
 		$result = $this->evaluate( 'otherPlugin::context.key' );
 		$this->assertEquals( 'otherPlugin-context', $result );
+
+		$result = $this->evaluate( 'state.obj.prop' );
+		$this->assertSame( 'myPlugin-state', $result );
```

Comment:
This is interesting, I don't know what result we'll get here. I assume it's a string `1`. I'm not sure how the path is parsed and if it accounts for numeric access or coercion happens so that numeric string paths access arrays:

```suggestion
		$this->assertSame( 'myPlugin-state', $result );

		$result = $this->evaluate( 'state.arrAccess.1' );
		$this->assertSame( '1', $result );
```

Reply:
In this case, `state.arrAccess.1`, 1 is treated first a string. You can operate, in example 
```
public function offsetGet( $offset ): mixed {
						return 2 * $offset; }
```
And will return 2 as a number.

What would be the best approach here?

Reply:
I think it's fine as it. It's just interesting that `"1"` is used to access numeric arrays.